src/katsdpcontroller/tasks.py

```diff
@@ -439,8 +442,13 @@ async def issue_req(self, req, args=(), timeout=None):
             self.logger.info("Request %s %s to node %s successful", req, args, self.name)
             return (reply, informs)
         except (FailReply, InvalidReply, OSError, asyncio.TimeoutError) as error:
-            msg = f"Failed to issue req {req} to node {self.name}. {error}"
+            if isinstance(error, asyncio.TimeoutError):
+                error_msg = f"Timed out after {timeout} s."
+            else:
+                error_msg = str(error)
+            msg = f"Failed to issue req {req} to node {self.name}. {error_msg}"
             self.logger.warning("%s", msg)
+            self.mark_suspect()
```

Comment:
The PC does some validation, but it's a good point: if something slips through the validation in the PC but is actually malformed, one might end up marking the entire array as suspect. I'll rework it so that FailReply and InvalidReply don't cause the engine to be considered suspect.

Reply:
I've updated it.